util/releasenotes.py

```diff
@@ -0,0 +1,111 @@
+"""
+Copyright (C) 2018 Ocean Observatories Initiative.
+"""
+
+import re
+
+
+class ReleaseNotes:
+    """
+    Python Object representation of markdown formatted release notes.
+    """
+
+    # default value for the Release Notes file
+    DEFAULT_FILE = "RELEASE_NOTES.md"
+
+    # value for when the version cannot be determined
+    UNKNOWN_VERSION = "0.0.1-unknown"
+
+    # Matches the component name that should be the first header of the release notes
+    # for example: "# Stream Engine"
+    _COMPONENT_NAME_PATTERN = re.compile("#\s+(?P<name>.*)");
+
+    # Matches Patterns such as: "# Development Release 1.0.0 2018-11-01"
+    _VERSION_PATTERN = re.compile("#\s+(?P<descriptor>[\w\s]+)\s+(?P<version>\S+)\s+(?P<date>\S*)\s*")
+
+    # singleton instance to use so that file isn't double parsed
+    __instance = None
+
+    @staticmethod
+    def instance(file=DEFAULT_FILE):
+        """
+        :return: a single instance of this class
+        """
+        if ReleaseNotes.__instance == None:
+            ReleaseNotes.__instance = ReleaseNotes(file)
+        return ReleaseNotes.__instance
+
+    def __init__(self, file=DEFAULT_FILE):
+        self.file = file
+        self._contents = None
+        self._component_name = None
+        self._latest_descriptor = None
+        self._latest_version = ReleaseNotes.UNKNOWN_VERSION
+        self._latest_date = None
+        self._parsed = False
+
+    def contents(self):
+        """
+        :return: the contents of the file as a list of strings, or None if the contents couldn't
+        be read
+        """
+        if not self._contents:
+            with open(self.file, "r") as file_handle:
```

Comment:
FYI - single quotes work just as well in Python and you don't have to press the shift key. I just saved you countless Shift key presses. Your left pinky can thank me later.